tests/crypto_test.py

```diff
@@ -12,10 +12,81 @@
 import gpgme
 import mock
 
+from alot import crypto
+from alot.errors import GPGProblem, GPGCode
+
 from . import utilities
 
-from alot import crypto
-from alot.errors import GPGProblem
+
+MOD_CLEAN = utilities.ModuleCleanup()
+
+# A useful single fingerprint for tests that only care about one key. This
+# key will not be ambiguous
+FPR = "F74091D4133F87D56B5D343C1974EC55FBC2D660"
+
+# Some additional keys, these keys may be ambigiuos
+EXTRA_FPRS = [
+    "DD19862809A7573A74058FF255937AFBB156245D",
+    "2071E9C8DB4EF5466F4D233CF730DF92C4566CE7",
+]
+
+DEVNULL = open('/dev/null', 'w')
+MOD_CLEAN.add_cleanup(DEVNULL.close)
+
+
+@MOD_CLEAN.wrap_setup
+def setUpModule():
+    home = tempfile.mkdtemp()
+    MOD_CLEAN.add_cleanup(shutil.rmtree, home)
+    mock_home = mock.patch.dict(os.environ, {'GNUPGHOME': home})
+    mock_home.start()
+    MOD_CLEAN.add_cleanup(mock_home.stop)
+
+    # create a single context to use class wide.
+    ctx = gpgme.Context()
+    ctx.armor = True
+
+    # Add the public and private keys. They have no password
+    search_dir = os.path.join(os.path.dirname(__file__), 'static/gpg-keys')
+    for each in os.listdir(search_dir):
+        if os.path.splitext(each)[1] == '.gpg':
+            with open(os.path.join(search_dir, each)) as f:
+                ctx.import_(f)
+
+
+@MOD_CLEAN.wrap_teardown
+def tearDownModule():
+    # Kill any gpg-agent's that have been opened
+    lookfor = 'gpg-agent --homedir {}'.format(os.environ['GNUPGHOME'])
+
+    out = subprocess.check_output(['ps', 'xo', 'pid,cmd'], stderr=DEVNULL)
+    for each in out.strip().split('\n'):
+        pid, cmd = each.strip().split(' ', 1)
+        if cmd.startswith(lookfor):
+            subprocess.call(['kill', pid])
```

Comment:
we already make a lot of assumptions about being on *nix, but os.kill works with python 2.7. I'll use it instead.